src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -193,10 +194,16 @@ def _instrument_layer(self, layer, config):
         layer._forward = layer.module.forward
 
         def new_forward(*inputs):
+            mask = self.calc_mask(layer, config)
             # apply mask to weight
             old_weight = layer.module.weight.data
-            mask = self.calc_mask(layer, config)
-            layer.module.weight.data = old_weight.mul(mask)
+            mask_weight = mask['weight']
```

Comment:
seem weird that there is weight in mask, maybe rename mask into something else?

Reply:
```mask['weight']``` means mask of the weight, it seems straightforward